simapp/v2/simdv2/cmd/root_di.go

```diff
@@ -29,70 +31,157 @@ import (
 )
 
 // NewRootCmd creates a new root command for simd. It is called once in the main function.
-func NewRootCmd[T transaction.Tx]() *cobra.Command {
+func NewRootCmd[T transaction.Tx](args ...string) (*cobra.Command, error) {
 	var (
 		autoCliOpts   autocli.AppOptions
 		moduleManager *runtime.MM[T]
 		clientCtx     client.Context
 	)
-
-	if err := depinject.Inject(
-		depinject.Configs(
-			simapp.AppConfig(),
-			depinject.Provide(
-				ProvideClientContext,
-				// inject desired account types:
-				multisigdepinject.ProvideAccount,
-				basedepinject.ProvideAccount,
-				lockupdepinject.ProvideAllLockupAccounts,
-
-				// provide base account options
-				basedepinject.ProvideSecp256K1PubKey),
-			depinject.Supply(log.NewNopLogger()),
-		),
-		&autoCliOpts,
-		&moduleManager,
-		&clientCtx,
-	); err != nil {
-		panic(err)
+	// parse config
+	defaultHomeDir, err := clientv2helpers.DefaultHomeDir(".simappv2")
+	if err != nil {
+		return nil, err
 	}
 
+	// initial bootstrap root command for config parsing
 	rootCmd := &cobra.Command{
-		Use:           "simd",
+		Use:           "simdv2",
 		Short:         "simulation app",
 		SilenceErrors: true,
-		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
+	}
+	serverv2.SetPersistentFlags(rootCmd.PersistentFlags(), defaultHomeDir)
+	// update the global viper with the root command's configuration
+	viper.SetEnvPrefix(clientv2helpers.EnvPrefix)
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_", "-", "_"))
+	viper.AutomaticEnv()
+
+	srv, err := initRootCmd[T](rootCmd, log.NewNopLogger(), commandDependencies[T]{})
+	if err != nil {
+		return nil, err
+	}
+	cmd, _, err := rootCmd.Traverse(args)
```

Comment:
There must be a way to abstract from L62 to L95, so the user doesn't need to set it every time